analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -183,11 +183,20 @@ private ProgramState ProcessDebugAssertBoolSymbol(ProgramState state, IOperation
         {
             ObjectConstraint constraint when constraint == ObjectConstraint.Null => new[] { context.SetOperationConstraint(BoolConstraint.True) },
             ObjectConstraint constraint when constraint == ObjectConstraint.NotNull => new[] { context.SetOperationConstraint(BoolConstraint.False) },
+            _ when ConversionOperandType(invocation.Arguments[0].ToArgument().Value) is { } type && !CanBeNull(type) => new[] { context.SetOperationConstraint(BoolConstraint.False) },
             _ when invocation.Arguments[0].TrackedSymbol() is { } argumentSymbol => new[]
             {
                         context.SetOperationConstraint(BoolConstraint.True).SetSymbolConstraint(argumentSymbol, ObjectConstraint.Null),
                         context.SetOperationConstraint(BoolConstraint.False).SetSymbolConstraint(argumentSymbol, ObjectConstraint.NotNull),
             },
             _ => new[] { context.State }
         };
+
+    private static ITypeSymbol ConversionOperandType(IOperation operation) =>
+        operation.Kind == OperationKindEx.Conversion
+            ? ConversionOperandType(IConversionOperationWrapper.FromOperation(operation).Operand)
+            : operation.Type;
```

Comment:
You can use `UnwrapConversion` instead:
https://github.com/SonarSource/sonar-dotnet/blob/f702246054981b5d137ec73dbb1072f752c38abc/analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs#L78-L85